prod/westeurope/internal/api/functions_pushnotifications/function_app_slot_staging/terragrunt.hcl

```diff
@@ -0,0 +1,117 @@
+}
+
+dependency "notification_storage_account" {
+  config_path = "../../storage_notifications/account"
+}
+
+# Include all settings from the root terragrunt.hcl file
+include {
+  path = find_in_parent_folders()
+}
+
+terraform {
+  source = "git::git@github.com:pagopa/io-infrastructure-modules-new.git//azurerm_function_app?ref=v2.1.34"
+}
+
+locals {
+  commonvars                   = read_terragrunt_config(find_in_parent_folders("commonvars.hcl"))
+  service_api_url              = local.commonvars.locals.service_api_url
+  app_insights_ips_west_europe = local.commonvars.locals.app_insights_ips_west_europe
+}
+
+inputs = {
+  name                = "staging"
+  resource_group_name = dependency.resource_group.outputs.resource_name
+
+  resources_prefix = {
+    function_app     = "fn3"
+    app_service_plan = "fn3"
+    storage_account  = "fn3"
+  }
+
+  app_service_plan_info = {
+    kind     = "elastic"
+    sku_tier = "ElasticPremium"
+    sku_size = "EP1"
+  }
+
+  runtime_version = "~3"
+
+  application_insights_instrumentation_key = dependency.application_insights.outputs.instrumentation_key
+
+  app_settings = {
+    FUNCTIONS_WORKER_RUNTIME       = "node"
+    WEBSITE_NODE_DEFAULT_VERSION   = "12.18.0"
+    WEBSITE_RUN_FROM_PACKAGE       = "1"
+    FUNCTIONS_WORKER_PROCESS_COUNT = 4
+    NODE_ENV                       = "production"
+
+    // Keepalive fields are all optionals
+    FETCH_KEEPALIVE_ENABLED             = "true"
+    FETCH_KEEPALIVE_SOCKET_ACTIVE_TTL   = "110000"
+    FETCH_KEEPALIVE_MAX_SOCKETS         = "40"
+    FETCH_KEEPALIVE_MAX_FREE_SOCKETS    = "10"
+    FETCH_KEEPALIVE_FREE_SOCKET_TIMEOUT = "30000"
+    FETCH_KEEPALIVE_TIMEOUT             = "60000"
+
+    // Endpoint for the legacy notification hub namespace
+    AZURE_NH_HUB_NAME                       = dependency.notification_hub.outputs.name
+    
+    // We do not want to handle production traffic for the moment, but still we want to be able to trigger the function for testing purpose
+    // Hence, we use a wrong queue name. To enable production traffic, just uncomment the original value
+    NOTIFICATIONS_QUEUE_NAME                = "wrong-notifications-queue-name" // dependency.notification_queue.outputs.name
+    NOTIFICATIONS_STORAGE_CONNECTION_STRING = dependency.notification_storage_account.outputs.primary_connection_string
+
+    SLOT_TASK_HUBNAME = "StagingTaskHub"
+
+    // Disable functions
+    "AzureWebJobs.HandleNHNotificationCall.Disabled" = "1"
+
+    # this app settings is required to solve the issue:
+    # https://github.com/terraform-providers/terraform-provider-azurerm/issues/10499
+    WEBSITE_CONTENTSHARE = "staging-content"
+  }
+
+  app_settings_secrets = {
+    key_vault_id = dependency.key_vault.outputs.id
+    map = {
+      AZURE_NH_ENDPOINT            = "common-AZURE-NH-ENDPOINT"
+    }
+  }
+
+  allowed_subnets = [
+    dependency.subnet.outputs.id,
+  ]
+
+  allowed_ips = local.app_insights_ips_west_europe
```

Comment:
```suggestion
  allowed_ips = []
```

Reply:
the web test runs on the production slot. Provided that you don't need to add appinsight ips in the restrictions